assets/src/components/drawer.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { PropsWithChildren, useState } from "react"
+import DrawerTab from "../components/drawerTab"
+import { joinClasses } from "../helpers/dom"
+
+interface Props {
+  open?: boolean
+  onToggleOpen?: () => void
+}
+
+const Drawer = ({
+  open = true,
+  onToggleOpen,
+  children,
+}: PropsWithChildren<Props>) => (
+  <>
+    <div
+      className={joinClasses([
+        "c-drawer",
+        ...(open
+          ? ["c-drawer--visible"]
+          : ["c-drawer--hidden", "u-hideable--hidden"]),
+      ])}
+    >
+      <DrawerTab
+        isVisible={open}
+        toggleVisibility={onToggleOpen || (() => {})}
+      />
+      {children}
+    </div>
+    <button onClick={onToggleOpen} className="c-drawer__backdrop-button" />
+  </>
+)
+
+interface DrawerWithStateProps {
+  startOpen?: boolean
+  onToggleOpen: Props["onToggleOpen"]
```

Comment:
Is this prop needed? It doesn't seem to be used

Reply:
Oh shoot this was supposed to be used, but now that you mention it, yeah let's just remove it, if the caller wants to know when this changes, then they can just use `<Drawer>`

Reply:
```suggestion
```